packages/cli/src/__tests__/main.test.tsx

```diff
@@ -0,0 +1,111 @@
+jest.mock('enquirer', () => ({
+  AutoComplete: jest.fn(),
+}));
+
+jest.mock('../presets', () => ({
+  mockedPreset: jest.fn(),
+}));
+
+import chalk from 'chalk';
+import { AutoComplete } from 'enquirer';
+
+import presets from '../presets';
+import main from '../main';
+
+describe('main', () => {
+  beforeEach(() => {
+    jest.spyOn(global.console, 'log').mockImplementation();
+    jest.spyOn(global.console, 'warn').mockImplementation();
+  });
+
+  afterEach(() => {
+    (global.console.log as jest.Mock).mockReset();
+    (global.console.warn as jest.Mock).mockReset();
+    ((AutoComplete as unknown) as jest.Mock).mockReset();
+    (presets.mockedPreset as jest.Mock).mockReset();
+  });
+
+  const setup = async (preset: string) => {
+    const cli = {
+      flags: {
+        preset,
+      },
+      input: [],
+      unnormalizedFlags: { preset },
+      pkg: {},
+      help: '',
+      showHelp: jest.fn(),
+      showVersion: jest.fn(),
+    };
+
+    await main(cli);
+
+    return {
+      cli,
+    };
+  };
+
+  it('should warn when invalid preset is provided as cli input', async () => {
+    const { cli } = await setup('invalid-using-cli');
+
+    expect(AutoComplete).not.toHaveBeenCalledTimes(1);
```

Comment:
do we know of a better way to test this outside of using mocks for global + auto complete? seems a bit wild 🤔

Reply:
hmm. 🤔 testing the behavior and not mocking things. I love this. But still testing cli is bit wild right now. Mocking is very safe in this case, but it will be wonderful if we can test the exact behaviour. But sometimes cli's introduce so many side effects and that might messed up our unit testing.
Can we skip for now? 😛 I ll keep this suggestion in mind. hehe. May be one open source library in future 😉

Reply:
np